src/components/ActionModal/ActionModal.tsx

```diff
@@ -0,0 +1,82 @@
+import { Dialog, Transition } from '@headlessui/react';
+import clsx from 'clsx';
+import { FC, Fragment, PropsWithChildren } from 'react';
+import styles from './ActionModal.module.css';
+import { Button } from '../../';
+import { opacityToClassName } from '../../utils/style';
+import { AllOrNone } from '../../utils/types';
+
+type Opacity = 'normal' | 'darker';
+
+type BaseProps = {
+  onClose: () => void;
+  onPrimaryAction?: () => void;
+  header?: string;
+  primaryActionLabel?: string;
```

Comment:
[question]
For example, do you have a use case for only non-primary, secondary actions?
I was wondering why primary is not required.

Reply:
This is a specification for cases where there is an action in the modal content.
I really want to make it required, but I have no choice but to make it optional to cover existing use cases. 😢

Reply:
I understood!